### ISSUE TITLE
Completions menu overrides my input on Enter

### ISSUE THREAD

**Issue by holyjak** [warpdotdev/Warp#2078]
### Describe the bug

I have "Open completions menu as you type" on. One problem is that if I do not use it and instead type the arguments myself then when I press Enter the menu deletes my typed input.

Example: I type `git switch -c whatever` press Enter and end up with `git switch -c --conflict` b/c --conflict was the first option in the menu. So I must remember to dismiss the menu with Esc.

It seems that the menu is not completely dumb and sometimes works as expected, e.g. when I type `git commit -m whatever` then the menu disappeares of its own.

### To Reproduce

type `git switch -c whatever` press Enter and you end up with `git switch -c --conflict` 

### Expected behaviour

completions should never override my input and should understand better that I do not want them. In the case of `git switch -c whatever|` where `|` is my cursor position, it should understand it makes no sense to insert anything there and should deactivate itself (until I type a space)

### Operating System

MacOS

### OS Version

13.0

### Shell Version

fish, version 3.5.1

### Warp Version

v0.2022.11.01.08.03.stable_00

### Does this block you from using Warp daily?

No

**Comment by dannyneira** [comment]
Hi @holyjak, Thanks for letting us know! We're sorry about this, this is not an ideal experience. So I'm struggling to reproduce this issue, do you find it intermittent for you? how often does it happen and is it only when your SSH into a remote machine or run commands locally? any pattern you can find in this error? It is possible you have some other fish plugin that's not compatible. Please see more on debugging your dot/config files here: https://docs.warp.dev/help/known-issues#debugging

https://user-images.githubusercontent.com/16809145/201153667-4ca89102-e39e-48b4-8b4c-1768434daecc.mp4

**Comment by holyjak** [comment]
No, this is happening every time but only for some commands, such as `git switch`. It happens localy.

I would be happy to try with a different shell but not sure how to do it. IF I run zsh or bash in my fish shell, the completions menu stops popping up.

I removed my ~/.config/fish/ and opened a new tab in Warp but it exhibited the same problem.

**Comment by holyjak** [comment]
Another, related problem: I use `z.fish` to jump to a historical path. Obviously Warp does not know it and when I type it tries to offer subdirectories as autocompletion, when I press Enter it replaces what I typed with the directory:
<img width="387" alt="warp-vs-z" src="https://user-images.githubusercontent.com/624958/201210600-4b40a7dd-b5f5-4272-9fa4-ed8c61b69a7d.png">

Here I and up with `z Applications/` :(

**Comment by holyjak** [comment]
This problem also happens when I leverage fish shell's autocompletions (those displayed in light-grey inline). Perhaps Warp does not notice I did input something because I just pressed ➡ ?

**Comment by dannyneira** [comment]
@holyjak Yes I think the fish completions competing with warp completions.

You can disable parts of your dotfiles just for Warp by using this conditional statement:
```sh
# Fish
if test "$TERM_PROGRAM" != "WarpTerminal"
# > What you want to disable here <
end
```

**Comment by lrgalego** [comment]
I'm having the same issue, many times a day with fish.

one example. type fast and hit enter

`git add .`

it deletes the `.` and does not execute the command.

If I think the `.` and enter again it works.
Or I need to hit esc to hide the suggestions before typing enter.

**Comment by andreiborisov** [comment]
I have the same issue, and I don't think it has anything to do with fish:

The problem is that auto completions are accepted on `Enter,` so if you type anything manually, you accept a completion instead of invoking a typed command. This is a design oversight and not a bug.

**Comment by justinclayton** [comment]
Agree with @andreiborisov, [this](https://github.com/warpdotdev/Warp/issues/2078#issuecomment-2130328588) is the meat of the issue. My most prominent example:

In ZSH there is an option called `autocd` which allows you to cd into a directory by only typing its name (I believe Oh My Zsh has this on by default.) So all day I'm doing a lot of `..` instead of `cd ..`.

Unfortunately, the completion menu's first result is `...`, so when I type `..<ENTER>`, I get sent back two directories instead of one. This one was actually pretty insidious until I slowed down enough to realize what was happening. Because of this I've had to disable "Open completions menu as you type" to regain my sanity.

Everything else about Warp is amazing by the way!
